contracts/token-registrars/StandardizedTokenRegistrar.sol

```diff
@@ -0,0 +1,159 @@
+
+import { Multicall } from '../utils/Multicall.sol';
+
+import { AddressBytesUtils } from '../libraries/AddressBytesUtils.sol';
+
+contract StandardizedTokenRegistrar is IStandardizedTokenRegistrar, ITokenManagerType, Multicall, Upgradable {
+    using AddressBytesUtils for bytes;
+    using AddressBytesUtils for address;
+    using SafeTokenTransfer for IStandardizedToken;
+
+    IInterchainTokenService public immutable service;
+    bytes32 public immutable chainNameHash;
+
+    bytes32 private constant CONTRACT_ID = keccak256('standardized-token-registrar');
+
+    struct DeployParams {
+        address deployer;
+        bytes distributor;
+        bytes operator;
+    }
+
+    mapping(bytes32 => DeployParams) public deploymentParameterMap;
+
+    bytes32 internal constant PREFIX_STANDARDIZED_TOKEN_SALT = keccak256('standardized-token-salt');
+
+    constructor(address interchainTokenServiceAddress) {
+        if (interchainTokenServiceAddress == address(0)) revert ZeroAddress();
+        service = IInterchainTokenService(interchainTokenServiceAddress);
+        string memory chainName_ = IInterchainTokenService(interchainTokenServiceAddress).remoteAddressValidator().chainName();
+        chainNameHash = keccak256(bytes(chainName_));
+    }
+
+    /**
+     * @notice Getter for the contract id.
+     */
+    function contractId() external pure returns (bytes32) {
+        return CONTRACT_ID;
+    }
+
+    function getStandardizedTokenSalt(address deployer, bytes32 salt) public view returns (bytes32) {
+        return keccak256(abi.encode(PREFIX_STANDARDIZED_TOKEN_SALT, chainNameHash, deployer, salt));
+    }
+
+    function getStandardizedTokenId(address deployer, bytes32 salt) public view returns (bytes32 tokenId) {
+        tokenId = service.getCustomTokenId(address(this), getStandardizedTokenSalt(deployer, salt));
+    }
+
+    function getStandardizedTokenAddress(address deployer, bytes32 salt) public view returns (address tokenAddress) {
+        tokenAddress = service.getStandardizedTokenAddress(getStandardizedTokenId(deployer, salt));
+    }
+
+    function deployStandardizedToken(
+        bytes32 salt,
+        string calldata name,
+        string calldata symbol,
+        uint8 decimals,
+        uint256 mintAmount,
+        address distributor
+    ) external payable {
+        address sender = msg.sender;
+        salt = getStandardizedTokenSalt(sender, salt);
+        bytes32 tokenId = service.getCustomTokenId(address(this), salt);
+
+        service.deployAndRegisterStandardizedToken(salt, name, symbol, decimals, mintAmount, distributor);
+        ITokenManager tokenManager = ITokenManager(service.getTokenManagerAddress(tokenId));
+        tokenManager.transferOperatorship(sender);
+
+        IStandardizedToken token = IStandardizedToken(service.getStandardizedTokenAddress(tokenId));
+        token.safeTransfer(sender, mintAmount);
+    }
+
+    function deployRemoteStandarizedToken(
```

Comment:
Can we combine the two deploy methods into one? The interchain address tracker has the current chain, so we can specialize the logic by comparing the chain name being passed. Reduces number of entrypoints. Perhaps we can simplify this in ITS itself.

Reply:
Similarly, we could combine deploy custom token manager into one method